proxy/pkg/metrics/node_metrics.go

```diff
@@ -298,15 +298,31 @@ func CreateCounterNodeMetric(metricFactory MetricFactory, nodeDescription string
 	return m, nil
 }
 
-func CreateHistogramNodeMetric(metricFactory MetricFactory, nodeDescription string, mn Metric, buckets []float64) (Histogram, error) {
+func CreateHistogramNodeMetric(metricFactory MetricFactory, nodeDescription string, mn Metric, buckets []float64, labels ...string) (Histogram, error) {
+	customLabels := make(map[string]string)
+	for i := 0; i < len(labels); i = i + 2 {
+		customLabels[labels[i]] = labels[i+1]
+	}
 	m, err := metricFactory.GetOrCreateHistogram(
-		mn.WithLabels(map[string]string{nodeLabel: nodeDescription}), buckets)
+		mn.WithLabels(map[string]string{nodeLabel: nodeDescription}).WithLabels(customLabels), buckets)
 	if err != nil {
 		return nil, err
 	}
 	return m, nil
 }
 
+func CreateHistogramNodeRequestDurationMetrics(metricFactory MetricFactory, nodeDescription string, mn Metric, buckets []float64) (map[string]Histogram, error) {
```

Comment:
Can you try replicating what we have on the `ProxyMetrics` side, i.e., splitting `OriginRequestDuration` into `OriginWritesDuration` and `OriginReadsDuration` (same for the target one)? Shouldn't that work and remove the need to have this new code here on these 2 functions?